debug_toolbar/middleware.py

```diff
@@ -88,6 +103,36 @@ def __call__(self, request):
             for panel in reversed(toolbar.enabled_panels):
                 panel.disable_instrumentation()
 
+        return self._postprocess(request, response, toolbar)
+
+    async def __acall__(self, request):
```

Comment:
Instead of `__acall__` what do you think about making the async checks a bit more targeted:

```python

    async def aget_response(self, request):
        return await self.get_response(request)

    async def aprocess_request(self, toolbar, request):
        return await toolbar.process_request(request)

    def __call__(self, request):
        # Decide whether the toolbar is active for this request.
        show_toolbar = get_show_toolbar()
        if not show_toolbar(request) or DebugToolbar.is_toolbar_request(request):
            return (
                self.aget_response(request)
                if self.async_mode
                else self.get_response(request)
            )

        toolbar = DebugToolbar(request, self.get_response)

        # Activate instrumentation ie. monkey-patch.
        for panel in toolbar.enabled_panels:
            panel.enable_instrumentation()
        try:
            # Run panels like Django middleware.
            response = (
                self.aprocess_request(toolbar, request)
                if self.async_mode
                else toolbar.process_request(request)
            )
```

cc: @matthiask

Reply:
This would get rid of `_postprocess` method and the duplicated logic between `call` and `acall`.

Reply:
But wouldn't calling async functions like `aget_response` and `aprocess_request` would just return the `coroutine` object rather than the actual value `HttpResponse` in this case, it needs to be awaited  in order to get the actual value that function returns and  awaiting can only be done in async function itself.

then we also need to change other panel methods like `generate_stats` and `generate_server_timing` etc and other parts which are working with response instance directly.

Reply:
Ah, you're right. That was would be a problem for process_request case. The get_response matches the __acall__ approach.

Reply:
Yeah, I can't think of a way around that. I'm not super thrilled about the `_process` design, but I don't have anything better at the moment. And we can always change it in the future. Go ahead and revert that change.